tests/core/data_model/test_base.py

```diff
@@ -161,6 +164,86 @@ def __init__(self, foo):
         assert issubclass(lib.Object, DataBase)
 
 
+def test_compiled_proto_init():
+    """Make sure that support for 'compiled' protos works cleanly without using
+    the dataobject wrapper
+    """
+    # pylint: disable=duplicate-code
+    with temp_data_model(
+        make_proto_def(
+            {
+                "ThingOne": {
+                    "foo": str,
+                    "bar": int,
+                }
+            },
+            mock_compiled=True,
```

Comment:
This makes the test simulate using a pre-compiled protobuf that uses `DataBase` directly without `@dataobject`.